sys/include/net/ng_netconf.h

```diff
@@ -0,0 +1,93 @@
+    NETCONF_OPT_IS_CHANNEL_CLR,     /**< check if channel is clear */
+    NETCONF_OPT_ADDRESS,            /**< get/set link layer address in host byte
+                                         order */
+    NETCONF_OPT_ADDRESS_LONG,       /**< long link layer address in host byte
+                                         order (e.g. IEEE802.15.4 EUI-64) */
+    NETCONF_OPT_ADDR_LEN,           /**< get the default address length a
+                                         network device expects as uint16_t in
+                                         host byte order */
+    NETCONF_OPT_NID,                /**< get/set the network ID (e.g. PAN ID in
+                                         IEEE 802.15.4) as uint16_t in host
+                                         byte order */
+    NETCONF_OPT_TX_POWER,           /**< get/set the output power for radio
+                                         devices in dBm as int16_t in host byte
+                                         order */
+    NETCONF_OPT_MAX_PACKET_SIZE,    /**< get/set the maximum packet size a
+                                         network module can handle as uint16_t
+                                         in host byte order */
+    NETCONF_OPT_PRELOADING,         /**< en/disable preloading or read the
+                                         current state. Preload using
+                                         *send_data*, send setting state to
+                                         *NETNETCONF_STATE_TX* */
+    NETCONF_OPT_PROMISCUOUSMODE,    /**< en/disable promiscuous mode or read
+                                         the current state */
+    NETCONF_OPT_AUTOACK,            /**< en/disable link layer auto ACKs or read
+                                         the current state */
+    NETCONF_OPT_PROTO,              /**< get/set the protocol for the layer
+                                         as type *ng_nettype_t*. */
+    NETCONF_OPT_STATE,              /**< get/set the state of network devices as
+                                         type *net_netconf_state_t* */
+    /* add more options if needed */
+} ng_netconf_opt_t;
+
+/**
+ * @brief   Binary parameter for enabling and disabling options
+ */
+typedef enum {
+    NETCONF_DISABLE = 0,            /**< disable a given option */
+    NETCONF_ENABLE = 1,             /**< enable a given option */
+} ng_netconf_enable_t;
+
+/**
+ * @brief   Option parameter to be used with *NETCONF_OPT_STATE* to set or get
+ *          the state of a network device
+ */
+typedef enum {
+    NETCONF_STATE_OFF = 0,          /**< device is powered off */
+    NETCONF_STATE_SLEEP,            /**< device is in sleep mode */
+    NETCONF_STATE_IDLE,             /**< device is in idle mode */
+    NETCONF_STATE_RX,               /**< device is in receive mode */
+    NETCONF_STATE_TX,               /**< device is in transmit mode */
+    /* add other states if needed */
+} ng_netconf_state_t;
```

Comment:
I'm with @haukepetersen, in theory you can apply all this states to a protocol's state machine, too.